packages/dataviews/src/dataform-controls/array.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * WordPress dependencies
+ */
+import { FormTokenField } from '@wordpress/components';
+import { useCallback, useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { DataFormControlProps } from '../types';
+
+export default function ArrayControl< Item >( {
+	data,
+	field,
+	onChange,
+	hideLabelFromVision,
+}: DataFormControlProps< Item > ) {
+	const { id, label, placeholder, suggestions, ...extraProps } = field;
```

Comment:
About `...extraProps`: if we need a field to configure a UI component, we may want to come with a structured (and safer) way to do it.

Reply:
Perfect, that makes sense.

Keeping this in mind, I went ahead and added a few defaults here https://github.com/WordPress/gutenberg/pull/71136/commits/f48d7e0785a7e56b4119045360e15580f09ab6b2

1. Make the selection open on focus, by default.
2. Make sure the user can only select from the elements provided.

I think in the case of a multiselect, this makes sense as we want to see the possible options (open on focus), and we shouldn't be able to add new elements (ie, the validation function).

Let me know if that sounds good to you. I'll look into failing tests next